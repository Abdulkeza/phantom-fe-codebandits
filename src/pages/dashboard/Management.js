import CheckRole from '@utils/CheckRoles.js';
import React from 'react';
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs';
import Buses from '../management/Buses.js';
import Companies from '../management/Companies.js';
import Drivers from '../management/Drivers.js';
import Operators from '../management/Operators.js';
import Routes from '../management/Routes.js';

const Management = () => {
  const allTabs = [
    { name: 'Drivers', roles: ['operator', 'admin', 'driver'], focus: true },
    { name: 'Companies', roles: ['admin'], focus: false },
    { name: 'Operators', roles: ['admin'], focus: false },
    { name: 'Buses', roles: ['operator', 'admin'], focus: false },
    { name: 'Routes', roles: ['admin', 'operator'], focus: false }
  ];
  return (
    <main className="overflow-x-auto">
      <Tabs selectedTabClassName={'border-b-2 border-primary h-full mb-0'}>
        <TabList
          className={'flex border-b-2 mb-5 w-full justify-around shadow-main'}
        >
          {allTabs.map((tab) => (
            <CheckRole
              key={tab.name}
              children={
                <Tab
                  className={
                    'mr-16 font-bold py-2 w-full px-3 text-center cursor-pointer outline-none focus:outline-none font-raleway'
                  }
                  key={tab.name}
                >
                  {tab.name}
                </Tab>
              }
              role={tab.roles}
            ></CheckRole>
          ))}
        </TabList>

        <CheckRole
          children={
            <TabPanel>
              <Drivers />
            </TabPanel>
          }
          role={['operator', 'admin', 'driver']}
        ></CheckRole>
        <CheckRole
          children={
            <TabPanel>
              <Companies />
            </TabPanel>
          }
          role={['admin']}
        ></CheckRole>
        <CheckRole
          children={
            <TabPanel>
              <Operators />
            </TabPanel>
          }
          role={['admin']}
        ></CheckRole>

        <CheckRole
          children={
            <TabPanel>
              <Buses />
            </TabPanel>
          }
          role={['operator', 'admin']}
        ></CheckRole>
        <CheckRole
          children={
            <TabPanel>
              <Routes />
            </TabPanel>
          }
          role={['admin', 'operator']}
        ></CheckRole>
      </Tabs>
    </main>
  );
};

export default Management;
