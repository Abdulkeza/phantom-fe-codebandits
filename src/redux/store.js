import { configureStore } from '@reduxjs/toolkit';
import { counterReducer } from './reducers/counterReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';
import { profileReducer } from './reducers/profileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer
  }
});
