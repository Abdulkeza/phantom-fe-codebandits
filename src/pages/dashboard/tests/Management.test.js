import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../../redux/store.js';
import Management from '../Management';

describe('Registration', () => {
  describe('Operators', () => {
    it('Should render the management page', () => {
      const elem = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <Management />
            </BrowserRouter>
          </Provider>
        )
        .toJSON();
      expect(elem).toMatchSnapshot();
    });
  });
});
