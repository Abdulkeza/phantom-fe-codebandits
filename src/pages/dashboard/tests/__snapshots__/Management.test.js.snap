// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Operators Should render the management page 1`] = `
<main>
  <h1
    className="font-black text-2xl pb-6"
  >
    Management
  </h1>
  <div
    className="react-tabs"
    data-rttabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="flex border-b-2 mb-5"
      role="tablist"
    >
      
      
      
      
      
    </ul>
    
    
    
    
    
  </div>
</main>
`;
